spec/rex/socket_spec.rb

```diff
@@ -84,27 +84,7 @@
     subject { described_class.getaddress('whatever') }
 
     before(:example) do
-      expect(Socket).to receive(:gethostbyname).and_return(['name', ['aliases'], response_afamily, *response_addresses])
-    end
-
-    context 'when ::Socket.gethostbyname returns IPv4 responses' do
-      let(:response_afamily) { Socket::AF_INET }
-      let(:response_addresses) { ["\x01\x01\x01\x01", "\x02\x02\x02\x02"] }
-
-      it { is_expected.to be_a(String) }
-      it "should return the first ASCII address" do
-        expect(subject).to eq "1.1.1.1"
-      end
-    end
-
-    context 'when ::Socket.gethostbyname returns IPv6 responses' do
-      let(:response_afamily) { Socket::AF_INET6 }
-      let(:response_addresses) { ["\xfe\x80"+("\x00"*13)+"\x01", "\xfe\x80"+("\x00"*13)+"\x02"] }
-
-      it { is_expected.to be_a(String) }
-      it "should return the first ASCII address" do
-        expect(subject).to eq "fe80::1"
-      end
```

Comment:
Will this break framework if these don't pass anymore?

Reply:
_My understanding_, is that `::Socket.gethostbyname` returned an array with several pieces of information, which inlcluded the byte address. Now we're using `::Addrinfo.getaddrinfo`, which only returns the ip address, so these tests are now redundant.

The tests that expect `"1.1.1.1" & "fe80::1"` to be returned are already handled in other places. (Line 112 & 124)

_But_, this comment made me realise I didn't change the context description on those two tests, so I'll get that sorted.

Reply:
I think these tests need updated similar to how `.getaddresses` was - rather than updated 👀

Reply:
Sure, I'll get those added back in and updated 👍